c10/core/SymInt.h

```diff
@@ -31,9 +29,52 @@ using SymIntNode = c10::intrusive_ptr<SymIntNodeImpl>;
 // named data_.
 class C10_API SymInt {
  public:
+  // TODO: this needs to only accept integers, not pointers
   /*implicit*/ SymInt(int64_t d) : data_(d){};
   SymInt() = default;
 
+  // TODO: these implementations are bad
+  SymInt(const SymInt& s) : data_(0) {
+    if (s.is_symbolic()) {
+      *this = SymInt::toSymInt(s.toSymIntNodeImpl());
+    } else {
+      data_ = s.data_;
+    }
+  }
+  SymInt(SymInt&& s) : data_(s.data_) {
+    s.data_ = 0;
```

Comment:
Is 0 a valid empty state for this?

Reply:
Yes, it represents zero integer, which is the default initialization for int